lib/OrdinaryDiffEqCore/src/solve.jl

```diff
@@ -162,8 +167,13 @@ function DiffEqBase.__init(
     p = prob.p
 
     # Get the control variables
+    # If alias kwarg is just default, use alias_u0, which is false by default, or is set by a kwarg to solve
+    # If alias_u0 is not nothing, use the alias_u0 provided by the user
+    if isnothing(alias.alias_u0)
+        alias = ODEAliases(alias_u0)
+    end
```

Comment:
This is breaking since it does not do what is documented in older versions if users just set `alias_u0`.

Reply:
If users set `alias_u0` in `solve`, and `ODEAliases.alias_u0` is `nothing`, it uses the `alias_u0` from the solve call.  So the only way that the `alias_u0` from the solve call isn't used is if an explicit `ODEAliases` with `ODEAliases.alias_u0` either true or false is put in the call to solve.